lib/pluginmanager/search.rb

```diff
@@ -0,0 +1,29 @@
+# encoding: utf-8
+require 'rubygems/spec_fetcher'
+require "pluginmanager/command"
+
+class LogStash::PluginManager::Search < LogStash::PluginManager::Command
+
+  parameter "[PATTERN]", "pattern to look for"
+
+  option "--author", "NAME", "Show only plugins authored by this name"
+
+  def execute
+    LogStash::Bundler.setup!({:without => [:build, :development]})
+    fetcher = Gem::SpecFetcher.fetcher
+    fetcher.detect(:latest) do |name_tuple|
+      File.fnmatch?(pattern, name_tuple.name)
```

Comment:
if the given pattern isn't bounded by asterisks, should we do it for them? My thinking is we should make this super-simple to use as a substring match that _happens_ to also support wildcards, so that a user could simply `bin/logstash-plugin search elasticsearch`.

~~~
pattern = '*' + pattern unless pattern.start_with?('*')
pattern = pattern + '*' unless pattern.end_with?('*')
~~~

Reply:
++, I honestly didn't concern myself too much with a polished UX (e.g. the result format is terrible, date format is showing time as well even though it's 00:00:00)

I'll add that on Monday